modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/UsersEndpoint.java

```diff
@@ -200,74 +193,79 @@ public Response getUsers(@QueryParam("filter") String filter, @QueryParam("sort"
       User user = i.next();
 
       // Filter list
-      if (filterName.isSome() && !filterName.get().equals(user.getName())
-              || (filterRole.isSome()
-                  && !Stream.$(user.getRoles()).map(getRoleName).toSet().contains(filterRole.get()))
-              || (filterProvider.isSome()
-                  && !filterProvider.get().equals(user.getProvider()))
-              || (filterText.isSome()
-                  && !TextFilter.match(filterText.get(), user.getUsername(), user.getName(), user.getEmail(), user.getProvider())
-                  && !TextFilter.match(filterText.get(), Stream.$(user.getRoles()).map(getRoleName).mkString(" ")))) {
+      final String finalFilterRole = filterRole;
```

Comment:
This is important since it declares the value as `final`. I couldn't use it later otherwise.

Reply:
Why could you not use the variable filterRole directly in lines 198/199?

Reply:
Because it is neither final nor effectively final and but Java requires that for use in lambdas, …

If you still do not believe me, please try it or read the [Java Language Specification](https://docs.oracle.com/javase/specs/jls/se10/html/jls-15.html#jls-15.27.2)

Reply:
Ok, thanks for the explanation.